packages/pigeon/platform_tests/test_plugin/windows/test_plugin.cpp

```diff
@@ -45,49 +46,101 @@ ErrorOr<AllTypes> TestPlugin::EchoAllTypes(const AllTypes& everything) {
   return everything;
 }
 
+ErrorOr<AllNullableTypes> TestPlugin::EchoAllNullableTypes(
+    const AllNullableTypes& everything) {
+  return everything;
+}
+
 std::optional<FlutterError> TestPlugin::ThrowError() {
   return FlutterError("An error");
 }
 
-ErrorOr<std::optional<std::string>> TestPlugin::ExtractNestedString(
-    const AllTypesWrapper& wrapper) {
-  const std::string* inner_string = wrapper.values().a_string();
+ErrorOr<int64_t> TestPlugin::EchoInt(int64_t an_int) { return an_int; }
+
+ErrorOr<double> TestPlugin::EchoDouble(double a_double) { return a_double; }
+
+ErrorOr<bool> TestPlugin::EchoBool(bool a_bool) { return a_bool; }
+
+ErrorOr<std::string> TestPlugin::EchoString(const std::string& a_string) {
+  return a_string;
+}
+
+ErrorOr<std::vector<uint8_t>> TestPlugin::EchoUint8List(
+    const std::vector<uint8_t>& a_uint8_list) {
+  return a_uint8_list;
+}
+
+ErrorOr<std::optional<std::string>> TestPlugin::ExtractNestedNullableString(
+    const AllNullableTypesWrapper& wrapper) {
+  const std::string* inner_string = wrapper.values().a_nullable_string();
   return inner_string ? std::optional<std::string>(*inner_string)
                       : std::nullopt;
 }
 
-ErrorOr<AllTypesWrapper> TestPlugin::CreateNestedString(
-    const std::string& string) {
-  AllTypes inner_object;
-  inner_object.set_a_string(string);
-  AllTypesWrapper wrapper;
+ErrorOr<AllNullableTypesWrapper> TestPlugin::CreateNestedNullableString(
+    const std::string* nullable_string) {
+  AllNullableTypes inner_object;
+  inner_object.set_a_nullable_string(*nullable_string);
+  AllNullableTypesWrapper wrapper;
   wrapper.set_values(inner_object);
   return wrapper;
 }
 
-ErrorOr<AllTypes> TestPlugin::SendMultipleTypes(bool a_bool, int64_t an_int,
-                                                const std::string& a_string) {
-  AllTypes someTypes;
-  someTypes.set_a_bool(a_bool);
-  someTypes.set_an_int(an_int);
-  someTypes.set_a_string(a_string);
+ErrorOr<AllNullableTypes> TestPlugin::SendMultipleNullableTypes(
+    const bool* a_nullable_bool, const int64_t* a_nullable_int,
+    const std::string* a_nullable_string) {
+  AllNullableTypes someTypes;
+  someTypes.set_a_nullable_bool(a_nullable_bool);
+  someTypes.set_a_nullable_int(a_nullable_int);
+  someTypes.set_a_nullable_string(*a_nullable_string);
   return someTypes;
 };
 
-ErrorOr<int64_t> TestPlugin::EchoInt(int64_t an_int) { return an_int; }
+ErrorOr<std::optional<int64_t>> TestPlugin::EchoNullableInt(
+    const int64_t* a_nullable_int) {
+  if (!a_nullable_int) {
+    return std::nullopt;
+  }
+  return *a_nullable_int;
+  *a_nullable_int;
```

Comment:
This is dead code. Same for the later methods.

Reply:
oops...